### ISSUE TITLE
BGFX Linux hangs on startup

### ISSUE THREAD

**Issue by francisdb** [vpinball/vpinball#1895]
10.8.1 branch
Ubuntu 24.04
Wayland

On startup the **bgfx** build hangs at:

```
2024-08-13 12:38:27.586 INFO  [213710] [ProgressDialog::SetProgress@217] Initializing Visuals... 10%
2024-08-13 12:38:27.586 INFO  [213710] [Player::Player@423] Initializing renderer (global states & resources)
2024-08-13 12:38:27.586 INFO  [213710] [RenderDevice::RenderDevice@413] BGFX available backends: Noop, Direct3D11, Direct3D12, OpenGL, Vulkan
```

**Comment by francisdb** [comment]
Stack:

```
__futex_abstimed_wait_common64 0x00007ffff1498d61
RenderDevice::RenderDevice RenderDevice.cpp:469
Renderer::Renderer Renderer.cpp:126
Player::Player player.cpp:431
PinTable::Play pintable.cpp:2524
VPinball::DoPlay vpinball.cpp:1074
VPApp::Run main.cpp:1122
WinMain main.cpp:1302
main main.cpp:1344
```

**Comment by jsm174** [comment]
Did a little debugging this morning and we will have to wait for @vbousquet:

Two issues:

1) BGFX using OpenGL on Linux has a shader compilation error:

```
2024-08-19 06:22:29.166 INFO  [115817] [tBGFXCallback::traceVargs@82] /home/jmillard/vpx-vpinball-10.8.1/standalone/linux-x64/tmp/bgfx.cmake/bgfx/src/renderer_gl.cpp (6901): BGFX 318 }
2024-08-19 06:22:29.166 ERROR [115817] [tBGFXCallback::fatal@57] /home/jmillard/vpx-vpinball-10.8.1/standalone/linux-x64/tmp/bgfx.cmake/bgfx/src/renderer_gl.cpp:6911BGFX FATAL 1: Failed to compile shader. 0: 0:257(29): error: syntax error, unexpected SAMPLE, expecting ')'
```

Line 257:

```
/home/jmillard/vpx-vpinball-10.8.1/standalone/linux-x64/tmp/bgfx.cmake/bgfx/src/renderer_gl.cpp (6901): BGFX 257 vec3 convertDmdToColor(vec4 sample)
2024-08-19 06:22:29.165 INFO  [115817] [tBGFXCallback::traceVargs@82] /home/jmillard/vpx-vpinball-10.8.1/standalone/linux-x64/tmp/bgfx.cmake/bgfx/src/renderer_gl.cpp (6901): BGFX 257 vec3 convertDmdToColor(vec4 sample)
/home/jmillard/vpx-vpinball-10.8.1/standalone/linux-x64/tmp/bgfx.cmake/bgfx/src/renderer_gl.cpp (6901): BGFX 258 {
```

2) BGFX locks up on Linux (Vulkan or OpenGL) at the `rd->m_frameReadySem.wait();`. Temporary hack was to sleep after the post.

 ``` if (!bgfx::init(init))
   {
      PLOGE << "BGFX initialization failed";
      exit(-1);
   }
   //bgfx::setDebug(BGFX_DEBUG_STATS);
   //bgfx::setDebug(BGFX_DEBUG_STATS | BGFX_DEBUG_WIREFRAME);
   const bool useVSync = init.resolution.reset & BGFX_RESET_VSYNC;
   bool vsync = useVSync;
   rd->m_frameReadySem.post();

   // Adding this for linux
   std::this_thread::sleep_for(std::chrono::milliseconds(500));
   
   while (rd->m_renderDeviceAlive)
   {
      // wait for a frame to be prepared by the logic thread
      rd->m_frameReadySem.wait();
      if (!rd->m_framePending)
         continue;
```

**Comment by francisdb** [comment]
Confirming the temp fix for xwayland. @vbousquet any idea?

**Comment by jsm174** [comment]
fwiw, this is also an issue on Android version of bgfx.

**Comment by mnlgg** [comment]
I just installed and tested the last version of VPX 8.1 BGFX over Ubuntu 22.04 and plays pretty well with most tables that I tried, using Vulkan! 

Only the Godzilla table didn’t work.

**Comment by francisdb** [comment]
Going to close this as this hack was implemented in #1960
